src/Entity/Transaction.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\TransactionRepository")
+ */
+class Transaction
+{
+    public const PAYBOX_SUCCESS = '00000';
+    public const PAYBOX_CONNECTION_FAILED = '00001';
+    public const PAYBOX_INTERNAL_ERROR = '00003';
+    public const PAYBOX_CARD_NUMBER_INVALID = '00004';
+    public const PAYBOX_CARD_END_DATE_INVALID = '00008';
+    public const PAYBOX_CARD_UNAUTHORIZED = '000021';
+    public const PAYBOX_PAYMENT_PAGE_TIMEOUT = '000030';
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="integer", options={"unsigned"=true})
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxResultCode;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxAuthorizationCode;
+
+    /**
+     * @var array|null
+     *
+     * @ORM\Column(type="json_array", nullable=true)
+     */
+    private $payboxPayload;
+
+    /**
+     * @var \DateTimeImmutable|null
+     *
+     * @ORM\Column(type="datetime_immutable", nullable=true)
+     */
+    private $payboxDateTime;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(unique=true, nullable=true)
+     */
+    private $payboxTransactionId;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(nullable=true)
+     */
+    private $payboxSubscriptionId;
+
+    /**
+     * @var \DateTimeImmutable
+     *
+     * @ORM\Column(type="datetime_immutable")
+     */
+    private $createdAt;
+
+    /**
+     * @var Donation
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Donation", inversedBy="transactions")
+     */
+    private $donation;
+
+    public function __construct(Donation $donation, array $payboxPayload)
+    {
+        $this->donation = $donation;
+        $this->createdAt = new \DateTimeImmutable();
+        $this->payboxPayload = $payboxPayload;
+        $this->payboxResultCode = $payboxPayload['result'];
+
+        if ($transaction = $payboxPayload['transaction'] ?: false) {
```

Comment:
double if for one action?
what about `$this->payboxTransactionId = $payboxPayload['transaction'] ?? false;`

Reply:
I think he does not want to set `$this->payboxTransactionId` with `false`.

What about `$this->payboxTransactionId = $payboxPayload['transaction'] ?: null;`

Don't use `??` if `$payboxPayload['transaction']` is a mandatory field.